datafusion/physical-plan/src/aggregates/mod.rs

```diff
@@ -1271,8 +1272,11 @@ mod tests {
     fn new_spill_ctx(batch_size: usize, max_memory: usize) -> Arc<TaskContext> {
         let session_config = SessionConfig::new().with_batch_size(batch_size);
         let runtime = Arc::new(
-            RuntimeEnv::new(RuntimeConfig::default().with_memory_limit(max_memory, 1.0))
-                .unwrap(),
+            RuntimeEnv::new(
```

Comment:
Is the reason this fixes the linked issue that a fair spill pool allocates memory in some pre-determined fashion rather than based on the actual memory usage (which can vary depending on the order in which operations are executed)?

Reply:
Hmm, I guess so though I'm not very sure. I just feel that for spilling it makes more sense to use `FairSpillPool` and looks like it can fix the test issue.